src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -126,14 +125,60 @@ def createStageOutCommand(self, sourcePFN, targetPFN, options=None, checksums=No
             echo "gfal-copy exit status: $EXIT_STATUS"
             if [[ $EXIT_STATUS != 0 ]]; then
                echo "ERROR: gfal-copy exited with $EXIT_STATUS"
+               echo "Actual command which failed: %s"
+               echo "Source PFN: %s"
+               echo "Target PFN %s"
                echo "Cleaning up failed file:"
                %s
             fi
             exit $EXIT_STATUS
-            """ % self.createRemoveFileCommand(targetPFN)
+            """ % copyCommand copyCommandDict['source'] copyCommandDict['destination'] self.createRemoveFileCommand(targetPFN)
```

Comment:
This line became a bit harder to read as well. Perhaps we could use print-function to create it?

Reply:
Sounds reasonable, I will provide a possible implementation with the print-function with the next commit